ultradns-rest/src/main/java/denominator/ultradns/UltraDNSRestZoneApi.java

```diff
@@ -0,0 +1,133 @@
+
+  @Inject
+  UltraDNSRestZoneApi(UltraDNSRest api) {
+    this.api = api;
+  }
+
+  /**
+   * in UltraDNSRest, zones are scoped to an account.
+   */
+  @Override
+  public Iterator<Zone> iterator() {
+    final String accountName = getCurrentAccountName();
+    List<String> zoneNames = new ArrayList<String>();
+    if (accountName != null) {
+      zoneNames = ZoneUtil.getZoneNames(api.getZonesOfAccount(accountName).getZones());
+    }
+    final Iterator<String> delegate = zoneNames.iterator();
+    return new Iterator<Zone>() {
+      @Override
+      public boolean hasNext() {
+        return delegate.hasNext();
+      }
+
+      @Override
+      public Zone next() {
+        return fromSOA(delegate.next());
+      }
+
+      @Override
+      public void remove() {
+        throw new UnsupportedOperationException();
+      }
+    };
+  }
+
+  @Override
+  public Iterator<Zone> iterateByName(String name) {
+    Zone zone = null;
+    try {
+      zone = fromSOA(name);
+    } catch (UltraDNSRestException e) {
+      if (e.code() != UltraDNSRestException.ZONE_NOT_FOUND
+          && e.code() != UltraDNSRestException.INVALID_ZONE_NAME) {
+        throw e;
+      }
+    }
+    return singletonIterator(zone);
+  }
+
+  @Override
+  public String put(Zone zone) {
+    try {
+      String accountName = getCurrentAccountName();
+      api.createPrimaryZone(zone.name(), accountName, "PRIMARY", false, "NEW");
+    } catch (UltraDNSRestException e) {
+      if (e.code() != UltraDNSRestException.ZONE_ALREADY_EXISTS) {
+        throw e;
+      }
+    }
+
+    RRSet soa = api.getResourceRecordsOfDNameByType(zone.name(), zone.name(),
+            ResourceTypes.SOA.code()).getRrSets().get(0);
+    soa.setTtl(zone.ttl());
+    List<String> rDataList = Arrays.asList(soa.getRdata().get(0).split("\\s"));
+    final int emailIndex = 1;
+    final int ttlIndex = 6;
+    rDataList.set(emailIndex, formatEmail(zone.email()));
+    rDataList.set(ttlIndex, String.valueOf(zone.ttl()));
+    List<String> newRDataList = new ArrayList<String>();
+    newRDataList.add(StringUtils.join(rDataList, " "));
+    soa.setRdata(newRDataList);
+    api.updateResourceRecord(zone.name(), ResourceTypes.SOA.code(), zone.name(), soa);
+
+    return zone.name();
+  }
+
+  @Override
+  public void delete(String name) {
+    try {
+      api.deleteZone(name);
+    } catch (UltraDNSRestException e) {
+      if (e.code() != UltraDNSRestException.ZONE_NOT_FOUND) {
+        throw e;
+      }
+    }
+  }
+
+  private Zone fromSOA(String name) {
+    List<Record> soas = RRSetUtil.buildRecords(api.getResourceRecordsOfDNameByType(name, name,
+            ResourceTypes.SOA.code()).rrSets());
+    checkState(!soas.isEmpty(), "SOA record for zone %s was not present", name);
+    Record soa = soas.get(0);
+    return Zone.create(name, name, soa.getTtl(), soa.getRdata().get(1));
+  }
+
+  private String getCurrentAccountName() {
+    AccountList accountList = api.getAccountsListOfUser();
```

Comment:
So, if a user have multiple accounts, this will create the new zone under the last account returned by GET accounts of user. We should have a new optional parameter in Netflix Denominator for the input account, so that user can provide the account under which he wants to create the zone. 
@sdiamond, please let us know your thoughts.
FYI: @vpavlyuk